sma/speedwire/speedwirediscovery.cpp

```diff
@@ -359,20 +209,20 @@ void SpeedwireDiscovery::processDatagram(const QHostAddress &senderAddress, quin
         if (m_inverters.contains(senderAddress)) {
             inverter = m_inverters.value(senderAddress);
         } else {
-            inverter = new SpeedwireInverter(senderAddress, Speedwire::sourceModelId(), m_localSerialNumber, this);
+            inverter = new SpeedwireInverter(m_speedwireInterface, senderAddress, Speedwire::sourceModelId(), m_localSerialNumber, this);
             m_inverters.insert(senderAddress, inverter);
         }
 
         SpeedwireInverterReply *reply = inverter->sendIdentifyRequest();
-        qCDebug(dcSma()) << "SpeedwireDiscovery: send identify request to" << senderAddress.toString();
-        connect(reply, &SpeedwireInverterReply::finished, this, [this, inverter, senderAddress, reply](){
-            qCDebug(dcSma()) << "SpeedwireDiscovery: identify request finished from" << senderAddress.toString() << reply->error();
-
-            SpeedwireInverterReply *loginReply = inverter->sendLoginRequest();
-            qCDebug(dcSma()) << "SpeedwireDiscovery: make login attempt using the default password.";
-            connect(loginReply, &SpeedwireInverterReply::finished, this, [loginReply, senderAddress](){
-                qCDebug(dcSma()) << "SpeedwireDiscovery: login attempt finished" << senderAddress.toString() << loginReply->error();
-            });
+        qCDebug(dcSma()) << "SpeedwireDiscovery: Send identify request to" << senderAddress.toString();
+        connect(reply, &SpeedwireInverterReply::finished, this, [/*this, inverter,*/ senderAddress, reply](){
+            qCDebug(dcSma()) << "SpeedwireDiscovery: Identify request finished from" << senderAddress.toString() << reply->error();
+
+//            SpeedwireInverterReply *loginReply = inverter->sendLoginRequest();
+//            qCDebug(dcSma()) << "SpeedwireDiscovery: make login attempt using the default password.";
+//            connect(loginReply, &SpeedwireInverterReply::finished, this, [loginReply, senderAddress](){
+//                qCDebug(dcSma()) << "SpeedwireDiscovery: login attempt finished" << senderAddress.toString() << loginReply->error();
+//            });
```

Comment:
Hmm, this doesn't seem to do anything, which probably is why it's commented out, but the whole block after sendIdentityRequest() doesn't seem to do anything either. Is this just for the debug prints?